src/mono/mono/metadata/object-internals.h

```diff
@@ -2033,7 +2033,7 @@ mono_gchandle_new_internal (MonoObject *obj, mono_bool pinned);
 MONO_COMPONENT_API MonoGCHandle
 mono_gchandle_new_weakref_internal (MonoObject *obj, mono_bool track_resurrection);
 
-MONO_COMPONENT_API ICALL_EXTERN_C
+MONO_COMPONENT_API ICALL_EXPORT
```

Comment:
This will duplicate MONO_API, I believe we should just pick on of them and not both.

Reply:
Since we always needs to export it regardless of  ENABLE_ICALL_EXPORT I believe it should only use MONO_COMPONENT_API.